packages/pluginutils/src/dataToEsm.ts

```diff
@@ -4,8 +4,8 @@ import makeLegalIdentifier from './makeLegalIdentifier';
 
 export type Indent = string | null | undefined;
 
-function stringify(obj: unknown): string {
-  return (JSON.stringify(obj) || 'undefined').replace(
+function stringify(val: boolean | number | string): string {
```

Comment:
why is this change necessary?

Reply:
because rest types (`undefined` `object` `symbol` `bigint`) already and should be handled in `serialize`, I think this will make the code more strong.

if it's not necessary, this change could be reverted.

Reply:
I reverted this change for CI undue error.